src/cephadm/cephadm

```diff
@@ -5212,6 +5257,462 @@ def command_gather_facts():
 ##################################
 
 
+class CephadmDaemonHandler(BaseHTTPRequestHandler):
```

Comment:
I would like more to have only one class for the cephdaemon http server.
I think it is more clear if you move the code from all the code from `CephadmDaemonHandler` to `CephadmHTTPServer` and remove the class `CephadmDaemonHandler`

```
class CephadmHTTPServer ( ThreadingMixIn ):
    all the stuff here... ( mypy will be also more happy)
```

Reply:
The instance creation for the http.server instance requires a request handler class. Every request then causes an instance of the handler class - so I don't think we can simplify to the extent your talking about...